pinot-batch-ingestion/pinot-standalone/src/main/java/org/apache/pinot/ingestion/standalone/SegmentTarPushJobRunner.java

```diff
@@ -0,0 +1,118 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.ingestion.standalone;
+
+import com.google.common.base.Preconditions;
+import java.io.File;
+import java.io.InputStream;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.apache.commons.configuration.Configuration;
+import org.apache.commons.configuration.MapConfiguration;
+import org.apache.pinot.common.utils.FileUploadDownloadClient;
+import org.apache.pinot.common.utils.SimpleHttpResponse;
+import org.apache.pinot.common.utils.retry.AttemptsExceededException;
+import org.apache.pinot.common.utils.retry.RetriableOperationException;
+import org.apache.pinot.common.utils.retry.RetryPolicies;
+import org.apache.pinot.filesystem.PinotFSFactory;
+import org.apache.pinot.ingestion.common.Constants;
+import org.apache.pinot.ingestion.common.PinotClusterSpec;
+import org.apache.pinot.ingestion.common.PinotFSSpec;
+import org.apache.pinot.ingestion.common.SegmentGenerationJobSpec;
+import org.apache.pinot.spi.filesystem.PinotFS;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class SegmentTarPushJobRunner {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentTarPushJobRunner.class);
+
+  private SegmentGenerationJobSpec _spec;
+
+  public SegmentTarPushJobRunner(SegmentGenerationJobSpec spec) {
+    _spec = spec;
+  }
+
+  public void run()
+      throws Exception {
+    //init all file systems
+    List<PinotFSSpec> pinotFSSpecs = _spec.getPinotFSSpecs();
+    for (PinotFSSpec pinotFSSpec : pinotFSSpecs) {
+      Configuration config = new MapConfiguration(pinotFSSpec.getConfigs());
+      PinotFSFactory.register(pinotFSSpec.getScheme(), pinotFSSpec.getClassName(), config);
+    }
+
+    //Get outputFS for writing output pinot segments
+    URI outputDirURI = new URI(_spec.getOutputDirURI());
+    PinotFS outputDirFS = PinotFSFactory.create(outputDirURI.getScheme());
+    outputDirFS.mkdir(outputDirURI);
+
+    //Get list of files to process
+    String[] files = outputDirFS.listFiles(outputDirURI, true);
+
+    List<String> segmentsToPush = new ArrayList<>();
+    for (String file : files) {
+      if (file.endsWith(Constants.TAR_GZ_FILE_EXT)) {
+        segmentsToPush.add(file);
+      }
+    }
+    pushSegments(outputDirFS, segmentsToPush);
+  }
+
+  public void pushSegments(PinotFS fileSystem, List<String> tarFilePaths)
+      throws RetriableOperationException, AttemptsExceededException {
+    LOGGER.info("Start pushing segments: {} to locations: {}", tarFilePaths,
```

Comment:
Limit size of log to (say) 5 segments. We don't want to see a page worth of log, right?